client/nginx.go

```diff
@@ -535,7 +534,7 @@ func versionSupported(n int) bool {
 }
 
 func getAPIVersions(httpClient *http.Client, endpoint string) (*versions, error) {
-	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	ctx, cancel := context.WithTimeout(context.Background(), httpClient.Timeout)
```

Comment:
How many downstream projects use nginx-plus-go-client? Can we be sure this change is non-breaking for them?

Reply:
Default [Go http client do not specify Timeout](https://github.com/golang/go/blob/213504e5431891e8421a92ab72ac08981ce07516/src/net/http/client.go#L91L107) (means no timeout), therefore it should not be used in a prod env. Whoever uses this library needs to pass his own http client and remember to explicitly setup Timeout.


One option to reduce required "paperwork" from a user of the lib is to provide a min default http client with specified timeout, and ability to use a func option if the user wants to configure NGINX plus client with a custom http client.